web/src/user_profile.js

```diff
@@ -385,6 +386,7 @@ export function show_user_profile(user, default_tab_key = "profile-tab") {
         args.bot_type = settings_data.bot_type_id_to_string(user.bot_type);
     }
 
+    user_profile_subscribe_widget = undefined;
```

Comment:
This feels like the wrong place to clear this state. Usually one does this in an `on_hide` method passed into `modals.open`, so that it is always cleared when the modal is closed. 

We should also move this `user_streams_list_widget` clearing logic to the new `on_user_profile_hide` function used by this; these should happen at the same time, but nothing guarantees clearly that `hide_user_profile` is necessarily called every time the modal closes, whereas that would be true for an `on_hide` callback to the modal library.

```
export function hide_user_profile() {                                                                                   
    user_streams_list_widget = undefined;                                                                               
    modals.close_if_open("user-profile-modal");                                                                         
}                                                                                                                       
```

Reply:
@timabbott , done